### ISSUE TITLE
if using a credential the font is installed in the $env:localappdata\Microsoft\Windows\Fonts not $Env:windir\Fonts

### ISSUE THREAD

**Issue by brwilkinson** [mkht/DSCR_Font#5]
Hey i just wanted to say thank you for your github DSC Resources. You have done a great service sharing them.

I am using your AppX package resource and i am just looking at the Fonts resource.

I noticed that if I pass in a user credential, then the font gets installed in the user profile

```powershell
ls $env:LOCALAPPDATA\Microsoft\Windows\Fonts


    Directory: C:\Users\benwilk\AppData\Local\Microsoft\Windows\Fonts


Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----         2/27/2021   4:41 PM        1129160 Fira Code Light Nerd Font Complete.ttf
```

However the TEST of the Resource only checks in the $FontFolder = Join-Path $Env:windir '\Fonts'

So currently the User credential for the install is not supported. So I just figured I would let you know.

In this case, you would get a prompt that blocks DSC.

![image](https://user-images.githubusercontent.com/3605266/111041983-d80bb380-83ef-11eb-9a0b-25909b2dd634.png)

Now that I know this I will remove the credential and cleanup the font in the user directory.

**Comment by brwilkinson** [comment]
so I was thinking... something like this...

0.6.0\DSCResources\cFont\cFont.psm1   line 94

```powershell
    # https://docs.microsoft.com/en-us/powershell/scripting/dsc/resources/authoringresourcecomposite?view=powershell-7.1#supporting-psdscrunascredential
    if ($PsDscContext.RunAsUser) 
    {
        $FontFolder = Join-Path $Env:LOCALAPPDATA '\Microsoft\Windows\Fonts'
    }
    else
    {
        $FontFolder = Join-Path $Env:windir '\Fonts'
    }
```

**Comment by mkht** [comment]
Hi @brwilkinson 
Thank you for your reporting. And I am very happy that you are using my resources.

To be honest, I don't use this resource myself anymore, because Windows 10 1809 has changed the behavior regarding font installation so much that it no longer works properly.

I would like to work on fixing this issue.
I will probably need to take a different approach than the solution you provided and will need to completely rewrite the internal process.
Please give me some time.

**Comment by brwilkinson** [comment]
hey @mkht  aaah good to know. yeah I discovered the same thing. it was just hanging on the install  :)

I thought it worked as the user one time, however when I tried it again later, I couldn't get it working.

So I guess will have to look for an alternative.

**Comment by mkht** [comment]
We have released the [v0.7.0](https://www.powershellgallery.com/packages/DSCR_Font/0.7.0), which has been modified to work on Windows 10 1809 or later.

A new `FontName` parameter has been added for the fix. This parameter is **required**.
Once you have installed the font manually, please refer to `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Fonts` registry key to verify the exact font registration name.

![無題](https://user-images.githubusercontent.com/25482642/111062299-6749c180-84eb-11eb-8cef-21451d370c40.png)

I have tested the following configuration and confirmed that it works fine.
```PowerShell
$output = 'C:\MOF'
Configuration DSCR_Font_Sample
{
    Import-DscResource -ModuleName DSCR_Font
    Node localhost
    {
        cFont Add_Font_Sample
        {
            Ensure   = 'Present'
            FontName = 'Fira Code Light Nerd Font Complete (TrueType)'
            FontFile = 'C:\temp\Fira Code Light Nerd Font Complete.ttf'
        }
    }
}

DSCR_Font_Sample -OutputPath $output
Start-DscConfiguration -Path  $output -Verbose -Wait -Force
```

If you want the fonts to be installed only for a specific user, you can also use the `PSDscRunAsCredential` parameter.
```PowerShell
cFont Add_Font_Sample
{
    Ensure   = 'Present'
    FontName = 'Fira Code Light Nerd Font Complete (TrueType)'
    FontFile = 'C:\temp\Fira Code Light Nerd Font Complete.ttf'
    PsDscRunAsCredential = $UserCredential
}
```

Please try it out.
